lgc/builder/llpcYCbCrAddressHandler.cpp

```diff
@@ -0,0 +1,193 @@
+/*
+ ***********************************************************************************************************************
+ *
+ *  Copyright (c) 2020 Advanced Micro Devices, Inc. All Rights Reserved.
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ **********************************************************************************************************************/
+/**
+***********************************************************************************************************************
+* @file  llpcYCbCrAddressHandler.cpp
+* @brief LLPC source file: Implementation of LLPC class YCbCrAddressHandler
+***********************************************************************************************************************
+*/
+#include "llpcYCbCrAddressHandler.h"
+#include "llpcGfxRegHandler.h"
+#include "llpcInternal.h"
+
+#include "llvm/IR/Intrinsics.h"
+#include "llvm/IR/IntrinsicsAMDGPU.h"
+
+using namespace lgc;
+using namespace llvm;
+
+// =====================================================================================================================
+// The class is used for calculating and mantain the base address of each plane in YCbCr image.
+// Note: There are at most 3 planes with plane numbers:0, 1 and 2. If input planeNum == 1, it'll generate
+// the base address for plane 0.
+void YCbCrAddressHandler::GenBaseAddress(
+    uint32_t planeNum) // The number of planes
+{
+    // For YCbCr, the possible plane numbers are among 1 and 3.
```

Comment:
Above it says `[0, 2]`, here and in the assertion it's `[1, 3]`. Which range is it then?

Reply:
The input parameter "planeNum" is [1, 3], but the plane numbers are [0, 2], which means:
`planeNum == 1` => generate plane0
`planeNum == 2` => genreate plane0 and plane1
`planeNum == 3` => generate plane0, plane1 and plane2

This function accepts the concept of planNum rather a specific plane, because the calulation of plane n+1 is always based on plane n.

Reply:
OK, I see now.